src/utils/transactor.ts

```diff
@@ -0,0 +1,69 @@
+import { onCatch, Transactor, TransactorOptions } from 'hooks/Transactor'
+import { Contract } from '@ethersproject/contracts'
+import invariant from 'tiny-invariant'
+
+export function handleTransact({
+  args,
+  transactor,
+  contract,
+  fnName,
+  txOpts,
+  version,
+  optionalArgs = [],
+}: {
+  args: unknown[]
+  transactor?: Transactor
+  contract?: Contract
+  txOpts?: Omit<TransactorOptions, 'value'>
+  fnName: string
+  version: 'v1' | 'v2'
+  optionalArgs?: unknown[]
+}) {
+  // If transactor or contract is undefined
+  // go directly to catch
+
+  try {
+    invariant(transactor && contract, 'missing required')
+    const reqArgs = args.filter(arg => !optionalArgs.includes(arg))
+    const isMissingParameter =
+      args.length - optionalArgs.length !== reqArgs.length
+
+    if (isMissingParameter) {
+      throw new Error('missing parameter')
+    }
+
```

Comment:
I rebased and got a bunch of other commits in this branch.


I pushed this branch to more explicitly show work.


I was trying to get the error logic into a function that we could easily use across the board. 

While something like this works @wraeth-eth made a great point that once we start using juice-sdk a lot of this code is going to go away and we're not going to need it.


In the short term, I'm just going to keep fixing these hanging pending states more explicitly in transactor hook calls.


In the long run, I'm sure we'll figure out a better way of write this logic with juice-sdk.

Reply:
cc: @jbx-protocol/frontend-core